src/inmanta/plugins.py

```diff
@@ -345,50 +392,46 @@ def check_args(self, args: List[object], kwargs: Dict[str, object]) -> bool:
         Check if the arguments of the call match the function signature
         """
         max_arg = len(self.arguments)
-        required_args = [x[0] for x in self.arguments if len(x) == 2]
-
         if len(args) + len(kwargs) > max_arg:
             raise Exception(
                 "Incorrect number of arguments for %s. Expected at most %d, got %d"
                 % (self.get_signature(), max_arg, len(args) + len(kwargs))
             )
-        present_kwargs: FrozenSet[str] = frozenset(kwargs.keys())
         # check for missing arguments
-        if len(args) < len(required_args):
-            required_kwargs: FrozenSet[str] = frozenset(arg[0] for arg in self.arguments[len(args) : len(required_args)])
-            if not required_kwargs.issubset(present_kwargs):
-                missing: FrozenSet[str] = required_kwargs.difference(present_kwargs)
-                raise RuntimeException(
-                    None,
-                    "Missing %d required arguments for %s(): %s"
-                    % (len(missing), self.__class__.__function_name__, ",".join(missing)),
-                )
-        present_positional_args: FrozenSet[str] = frozenset(arg[0] for arg in self.arguments[: len(args)])
+        required_args: FrozenSet[str] = frozenset(arg.arg_name for arg in self.arguments if not arg.has_default_value())
+        present_positional_args: FrozenSet[str] = frozenset(arg.arg_name for arg in self.arguments[: len(args)])
+        present_kwargs: FrozenSet[str] = frozenset(kwargs.keys())
+        missing_args = required_args.difference({*present_positional_args, *present_kwargs})
+        if missing_args:
+            raise RuntimeException(
+                None,
+                "Missing %d required arguments for %s(): %s"
+                % (len(missing_args), self.__class__.__function_name__, ",".join(missing_args)),
+            )
         # check for kwargs overlap with positional arguments
-        if not present_kwargs.isdisjoint(present_positional_args):
+        overlapping_args: FrozenSet[str] = present_kwargs.intersection(present_positional_args)
+        if overlapping_args:
             raise RuntimeException(
                 None,
-                "Multiple values for %s in %s()"
-                % (",".join(present_kwargs.intersection(present_positional_args)), self.__class__.__function_name__),
+                "Multiple values for %s in %s()" % (",".join(overlapping_args), self.__class__.__function_name__),
```

Comment:
this could also be tested

Reply:
This is already tested here: https://github.com/inmanta/inmanta-core/blob/ca9b1725b12bc8fca4aa9f9fcdf33a4b6549d29b/tests/compiler/test_plugins.py#L87